src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -0,0 +1,330 @@
+#
+# SMB.CONF(5)		The configuration file for the Samba suite 
+# $FreeBSD$
+#
+<%
+        import os
+        import sys
+        sys.path.extend([
+            '/usr/local/www',
+            '/usr/local/www/freenasUI'
+        ])
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from freenasUI.common.freenassysctl import freenas_sysctl as fs
+        parsed_conf = {}
+
+        def get_db_config():
+            conf = {}
+
+            conf['cifs'] = middleware.call_sync('datastore.query', 'services.cifs', None, {'get': True})
```

Comment:
We would like to avoid using datastore whenever possible. Reason is that it creates another layer which data can be transformed and massaged for general use, not relying on what is on disk. One of the main advantages is getting rid of useless prefix `cifs_srv_`.

For SMB we have: `smb.config`.
For network config: `network.configuration.config`

And eventually replace ad/ldap/nis once we have them there as well.

Reply:
Thanks for the tips! I will make sure I minimize usage of datastore.